src/ImageSharp.Web/Caching/PhysicalFileSystemCache.cs

```diff
@@ -72,7 +72,10 @@ public class PhysicalFileSystemCache : IImageCache
 
             // Allow configuration of the cache without having to register everything.
             this.cacheOptions = cacheOptions != null ? cacheOptions.Value : new PhysicalFileSystemCacheOptions();
-            this.cacheRootPath = Path.Combine(environment.WebRootPath, this.cacheOptions.CacheFolder);
+            var cacheRoot = string.IsNullOrEmpty(this.cacheOptions.CacheRoot)
```

Comment:
Nice and none breaking 👍 I wonder if we can tweak it to allow both relative and absolute paths?

We'll need to add some unit tests also to ensure coverage.

Reply:
And relativ to which path? To the WebRoot? Any suggestions how to accomplish that?

I can look into tests.

Reply:
I would have added some tests. But creating a unit test for this is not so straight forward. I could only imagine to use a package like system.io.abstractions. But that will probably not help for the PhysicalFileProvider..

Reply:
I see only one way. We need to get rid of the instantiation of PhysicalFileProvider in the ctor. 

I would suggest to create a new interface for the FileAccess and a implementation which depends on the PhysicalFileProvider. So we can go and test the logic of setting the root of the Cache.

The question for the answer for the relative path is relatively easy 😄 ...

When you have a look at the docs you see that relative paths are not possible
https://docs.microsoft.com/de-de/aspnet/core/fundamentals/file-providers?view=aspnetcore-3.1#physicalfileprovider


So i am waiting for your suggestions how to proceed

Reply:
I think it's possible with the current setup. 

You can use `Path.IsPathFullyQualified` to check if the value is absolute. If it is, take it, if not use `Path.GetFullPath` to combine the relative path with the environment `ContentRootPath` absolute path

Reply:
i have added some test. SHould i look into the FullyQualified path in this MR or should we tackle this in a different one?

Reply:
Let's do it in one. There should be very little to add and we can update your new tests (nice!) to use inline theory data and test both scenarios.

Reply:
ok just for clarification. When specifying e.g. "./testfolder/xyz/" as CacheFolder in the options. You want it to be relative to webroot?

Reply:
I'd say relative to the application.